pegasus/sites.v3/hourofcode.com/views/map.haml

```diff
@@ -126,7 +129,23 @@
     var map_zoom = gmap.getZoom();
     var map_center = gmap.getCenter();
 
-    var map_height = map_width / 2;
+    // Max height of map is 2/3 of screen height so user can always scroll.
+    var max_height = $(window).innerHeight() * .65; 
+
+    var window_aspect_ratio = $(window).innerHeight() / $(window).innerWidth();
+    
+    if  (window_aspect_ratio < 1) {
+      // Landscape window. Use the current 1:2 ratio map size.
+      map_height = map_width / 2;
+
+      // But, make sure this size doesn't exceed our max height.
+       map_height = Math.min(map_height, max_height);
+    } 
+    else {
+      // Portrait window. Just use the max height so that it's taller and more usable.
+      map_height = max_height;
+    }
+
```

Comment:
definitely moving into territory where a common utility class would be handy.  (we can put things like that in /shared)